src/main/java/me/dinowernli/grpc/prometheus/ServerMetrics.java

```diff
@@ -121,13 +126,23 @@ public void recordStreamMessageReceived(Metadata metadata) {
    * Only has any effect if monitoring is configured to include latency histograms. Otherwise, this
    * does nothing.
    */
-  public void recordLatency(double latencySec, Metadata metadata) {
+  public void recordLatency(double latencySec, Metadata metadata, Optional<Code> codeOpt) {
     if (!this.serverHandledLatencySeconds.isPresent()) {
       return;
     }
-    addLabels(
-            this.serverHandledLatencySeconds.get(), customLabels(metadata, labelHeaderKeys), method)
-        .observe(latencySec);
+
+    final List<String> allLabels = new ArrayList<String>();
+    allLabels.addAll(customLabels(metadata, labelHeaderKeys));
+    if (addCodeLabelToHistograms) {
+      allLabels.add(codeOpt.map(c -> c.toString()).orElse(""));
+    }
+
+    addLabels(this.serverHandledLatencySeconds.get(), allLabels, method).observe(latencySec);
+  }
+
+  @Deprecated
+  public void recordLatency(double latencySec, Metadata metadata) {
```

Comment:
Why do we need both `recordLatency` methods? Can we just have a single one that always takes the response code (non optional), and just ignores it if `addCodeLabelToHistograms` is false?

Reply:
Initially left the old one to minimize potential binary compatibility issues, since it is public, in dependencies like this:
```
app -> some-lib(uses recordLatency directly) -> java-grpc-prometheus:old_version
   \-> java-grpc-prometheus:new_version
```
But probably, it is worth to sacrifice this for the sake of code clarity:)
What do you think?

Reply:
Sorry, just noted that ServerMetrics is package-private, so, removed the old version of method)